doc/refman/RefMan-ssr.tex

```diff
@@ -0,0 +1,4671 @@
+the \ssrC{[: name ]} intro pattern (see section~\ref{ssec:intro},
+page~\pageref{ssec:introabstract}).
+In a goal like the following:
+\begin{lstlisting}
+ m : nat
+ abs : <hidden>
+ n : nat
+ =============
+ m < 5 + n
+\end{lstlisting}
+The tactic \ssrC{abstract: abs n} first generalizes the goal with respect to
+\ssrC{n} (that is not visible to the abstract constant \ssrC{abs}) and then
+assigns \ssrC{abs}.  The resulting goal is:
+\begin{lstlisting}
+ m : nat
+ n : nat
+ =============
+ m < 5 + n
+\end{lstlisting}
+Once this subgoal is closed, all other goals having \ssrC{abs} in their context
+see the type assigned to \ssrC{abs}.  In this case:
+\begin{lstlisting}
+ m : nat
+ abs : forall n, m < 5 + n
+\end{lstlisting}
+
+For a more detailed example the user should refer to section~\ref{sssec:have},
+page~\pageref{sec:havetransparent}.
+
+\subsection{Introduction}\label{ssec:intro}
+
+The application of a tactic to a given goal can generate
+(quantified) variables, assumptions, or definitions, which the user may want to
+\emph{introduce} as new facts, constants or defined constants, respectively.
+If the tactic splits the goal into several subgoals,
+each of them may require the introduction of different constants and facts.
+Furthermore it is very common to immediately decompose
+or rewrite with an assumption instead of adding it to the context,
+as the goal can often be simplified and even
+proved after this.
+
+All these operations are performed by the introduction tactical
+`\ssrC{=>}', whose general syntax is
+% liste indic�e cf explication. Mettre la version ^+ dans le synopsis
+\begin{displaymath}
+	{\tac}\ \ssrC{=>}\ \ssrN[1]{i-item} \dots \ssrN[n]{i-item}
+\end{displaymath}
+where ${\tac}$ can be any tactic, $n > 0$ and
+\begin{eqnarray*}
+  \ssrN{i-item}& ::=&
+    \ssrN{i-pattern} \;|\; \ssrN{s-item} \;|\; \ssrN{clear-switch} \;|\; \ssrC{/}\ssrN{term} \\
+  \ssrN{s-item}& ::=& \ssrC{/=} \;|\; \ssrC{//} \;|\; \ssrC{//=} \\
+  \ssrN{i-pattern}& ::=&
+           \ssrN{ident} \;|\; \ssrC{_} \;|\; \ssrC{?} \;|\; \ssrC{*}\;|\;
+           [\ssrN{occ-switch}]\ssrC{->} \;|\; [\ssrN{occ-switch}]\ssrC{<-} \;|\; \\
+	   &&      \ssrC{[} \ssrN[1]{i-item}^* \;\ssrC{|}\;\dots ;\ssrC{|}\;\ssrN[m]{i-item}^* \ssrC{]}
+	   \;|\;\ssrC{-}\;|\;\ssrC{[:}\ssrN{ident}^+\ssrC{]}
+\end{eqnarray*}
+%subject to the condition that at least every other $\ssrN{i-item}$ should
+%be an $\ssrN{i-pattern}$ (for any two consecutive $\ssrN[i]{i-item}\
+%\ssrN{i-item}_{i+1}$, either $\ssrN[i]{i-item}$ or $\ssrN{i-item}_{i+1}$ is an
+%$\ssrN{i-pattern}$).
+
+The `\ssrC{=>}' tactical first executes ${\tac}$, then the
+$\ssrN{i-item}$s, left to right, i.e., starting from $\ssrN[1]{i-item}$.  An
+$\ssrN{s-item}$ specifies a simplification operation; a $\ssrN{clear
+switch}$ specifies context pruning as in~\ref{ssec:discharge}. The
+$\ssrN{i-pattern}$s can be seen as a variant of \emph{intro patterns}:
```

Comment:
If you are referring to Coq's intro-patterns, maybe a reference to the appropriate section could be useful (for those who do not know what this is).